documentation/modules/auxiliary/scanner/http/nagios_xi_scanner.md

```diff
@@ -0,0 +1,105 @@
+## Vulnerable Application
+
+The module detects the version of Nagios XI applications and suggests matching exploit modules based on the version number.
+
+The module takes advantage of the `Msf::Exploit::Remote::HTTP::NagiosXi` mixin in order to
+authenticate to the target and obtain the version number, which is only revealed to authenticated users.
+
+When used to target a specific host, the module requires valid credentials for a Nagios XI account.
+These can be provided via `USERNAME` and `PASSWORD` options.
+Alternatively, it is possible to provide a specific Nagios XI version number via the `VERSION` option.
+In that case, the module simply suggests matching exploit modules and does not probe the target(s).
+
+The module currently supports the following exploit modules:
+- exploit/linux/http/nagios_xi_plugins_check_ping_authenticated_rce (CVE-2019-15949)
+- exploit/linux/http/nagios_xi_plugins_filename_authenticated_rce (CVE-2020-35578)
+- exploit/linux/http/nagios_xi_mibs_authenticated_rce (CVE-2020-5791)
+- exploit/linux/http/nagios_xi_snmptrap_authenticated_rce (CVE-2020-5792)
```

Comment:
There's a bunch of other existing modules:

* [ ] modules/exploits/linux/http/nagios_xi_authenticated_rce.rb
  * `< 5.6.6`
* [ ] modules/exploits/linux/http/nagios_xi_chained_rce_2_electric_boogaloo.rb
  * `5.2.6` to `5.4.12` (presumably this is an inclusive range)
* [ ] modules/exploits/linux/http/nagios_xi_chained_rce.rb
  * `<= 5.2.7`
* [ ] modules/exploits/linux/http/nagios_xi_magpie_debug.rb
  * `5.5.6` (presumably some earlier versions are also affected)
* [x] ~~modules/exploits/linux/misc/nagios_nrpe_arguments.rb~~
  * not applicable
* [x] ~~modules/exploits/unix/webapp/nagios3_history_cgi.rb~~
  * This probably pre-dates XI. The exploit released in 2012 and simply states "Nagios3".
* [x] ~~modules/exploits/unix/webapp/nagios3_statuswml_ping.rb~~
  * This probably pre-dates XI. The exploit released in 2009 and simply states "Nagios3".
* [x] ~~modules/exploits/unix/webapp/nagios_graph_explorer.rb~~
  * unclear which versions are likely to be affected as this exploits a plugin. It is also unclear if the plugin is installed by default.

If the scanner module only uses the Nagios XI version to check then checks for most of these modules could be added very easily.

Reply:
The latest commit adds all these modules ( apart from the legacy ones).